src/components/invoices/EditInvoicePaymentStatusDialog.tsx

```diff
@@ -0,0 +1,143 @@
+  InvoiceForUpdateInvoicePaymentStatusFragment,
+  InvoicePaymentStatusTypeEnum,
+  UpdateInvoiceInput,
+  useUpdateInvoicePaymentStatusMutation,
+} from '~/generated/graphql'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import { ComboBoxField } from '~/components/form'
+import { theme } from '~/styles'
+
+gql`
+  fragment InvoiceForUpdateInvoicePaymentStatus on Invoice {
+    id
+    paymentStatus
+  }
+
+  mutation updateInvoicePaymentStatus($input: UpdateInvoiceInput!) {
+    updateInvoice(input: $input) {
+      id
+      ...InvoiceForUpdateInvoicePaymentStatus
+    }
+  }
+
+  ${AllInvoiceDetailsForCustomerInvoiceDetailsFragmentDoc}
+`
+
+export interface UpdateInvoicePaymentStatusDialogRef {
+  openDialog: (invoice: InvoiceForUpdateInvoicePaymentStatusFragment) => unknown
+  closeDialog: () => unknown
+}
+
+export const UpdateInvoicePaymentStatusDialog = forwardRef<UpdateInvoicePaymentStatusDialogRef>(
+  (_, ref) => {
+    const { translate } = useInternationalization()
+    const dialogRef = useRef<DialogRef>(null)
+    const [invoice, setInvoice] = useState<InvoiceForUpdateInvoicePaymentStatusFragment>()
+    const [updateInvoice] = useUpdateInvoicePaymentStatusMutation({
+      onCompleted({ updateInvoice: updateInvoiceRes }) {
+        if (updateInvoiceRes?.id) {
+          addToast({
+            message: translate('text_63eba8c65a6c8043feee2a02'),
+            severity: 'success',
+          })
+        }
+      },
+    })
+    const formikProps = useFormik<Omit<UpdateInvoiceInput, 'id'>>({
+      initialValues: {
+        paymentStatus: invoice?.paymentStatus || undefined,
+      },
+      validationSchema: object().shape({
+        paymentStatus: string().required(''),
+      }),
+      validateOnMount: true,
+      enableReinitialize: true,
+      onSubmit: async (values, formikBag) => {
+        if (!invoice?.id) return
+
+        const res = await updateInvoice({
+          variables: {
+            input: {
+              id: invoice.id,
+              ...values,
+            },
+          },
+        })
+
+        if (res.data?.updateInvoice) {
+          dialogRef?.current?.closeDialog()
+          formikBag.resetForm()
+        }
+      },
+    })
+
+    useImperativeHandle(ref, () => ({
+      openDialog: (data) => {
+        setInvoice(data)
+        dialogRef.current?.openDialog()
+      },
+      closeDialog: () => dialogRef.current?.closeDialog(),
+    }))
+
+    return (
+      <Dialog
+        ref={dialogRef}
+        title={translate('text_63eba8c65a6c8043feee2a0d')}
+        description={translate('text_63eba8c65a6c8043feee2a0e')}
+        onClickAway={() => {
+          formikProps.resetForm()
+          formikProps.validateForm()
+        }}
+        actions={({ closeDialog }) => (
+          <>
+            <Button
+              variant="quaternary"
+              onClick={() => {
+                closeDialog()
+                formikProps.resetForm()
```

Comment:
no need for `validateForm` here ?

Reply:
Nope, modal is init with a given prop, so no matter what was there before. Even tho we remove the data and close, value is set onOpen

We also do not show errors bellow the input and it's not possible to submit without value